### PR TITLE
feat: introduction to publish lifecycle

### PR DESCRIPTION
**Pull request by asalem1**
This is an initial pass at adding the save draft state functionality to a notebook.

### CODE REVIEW

File: src/flows/components/header/index.tsx [hunk 1]
```diff
@@ -133,6 +141,53 @@ const FlowHeader: FC = () => {
...
+        .then(resp => {
+          if (!resp.length) {
+            return
+          }
+        })
+        .catch(error => {
+          console.error({error})
+          setPublishLoading(RemoteDataState.Error)
+        })
+      setPublishLoading(RemoteDataState.Done)
```
Comment (drdelambre) [inline src/flows/components/header/index.tsx:161]: i think you wanna move this into the last then or add an await to the fetch

Comment (asalem1) [reply src/flows/components/header/index.tsx:161]: totally. Working on the openapi stuff now

**Review by drdelambre — APPROVED** [review]
good job breaking up this part of the PR from the version navigation part

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 97 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=3769)

**Pull request merged**